src/libraries/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -152,19 +152,19 @@ public void SetAccessControl_FileStream_FileSecurity_InvalidArguments()
         [Fact]
         public void SetAccessControl_FileStream_FileSecurity_InvalidFileSecurityObject()
         {
-            using var directory = new TempDirectory();
+            using var directory = new TempAclDirectory();
             using var file = new TempFile(Path.Combine(directory.Path, "file.txt"));
-            using FileStream fileStream = File.Open(file.Path, FileMode.Append, FileAccess.Write, FileShare.None);
+            using FileStream fileStream = File.Open(file.Path, FileMode.Open, FileAccess.Write, FileShare.Delete);
```

Comment:
Can you verify if this still needs to be changed in order to successfully delete the file on dispose?

Reply:
I added Open, Append and OpenOrCreate as inline data.

Reply:
I don't think that's relevant for this scenario which is testing `ArgumentNullException`.